azurerm/resource_arm_virtual_machine.go

```diff
@@ -1598,3 +1626,23 @@ func resourceArmVirtualMachineStorageImageReferenceHash(v interface{}) int {
 	}
 	return hashcode.String(buf.String())
 }
+
+func resourceArmVirtualMachineGetManagedDiskInfo(disk *compute.ManagedDiskParameters, meta interface{}) (*compute.Disk, error) {
+	client := meta.(*ArmClient).diskClient
+	ctx := meta.(*ArmClient).StopContext
+
+	var diskInfo *compute.Disk
+	if disk != nil && disk.ID != nil {
+		diskID, err := parseAzureResourceID(*disk.ID)
+		if err != nil {
+			return nil, err
+		}
+		diskResp, err := client.Get(ctx, diskID.ResourceGroup, diskID.Path["disks"])
+		if err != nil {
+			return nil, err
```

Comment:
_minor_ for consistency purposes and in order to return a more useful error message - can we update this to be:

```
id, err := parseAzureResourceID(*disk.ID)
if err != nil {
  return nil, fmt.Errorf("Error parsing Disk ID %q: %+v", *disk.ID, err)
}

resourceGroup := id.ResourceGroup
name := id.Path["disks"]
diskResp, err := client.Get(ctx, resourceGroup, name)
if err != nil {
  return nil, fmt.Errorf("Error retrieving Disk %q (Resource Group %q): %+v", name, resourceGroup, err)
}
```